### PR TITLE
3DS Launcher API

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

- Add 3DS launcher API 
- Consolidate `performVerification` and `continuePerformVerification` with the addition of the `ThreeDSecureLauncher` the determination of whether to "continue" with the verification happens before launching now
- Remove some of the unsupported 3DS V1 logic to allow for simplified integration - the remaining V1 code cleanup will happen in a future PR
- **Note:** Method/callback/result object names have not been updated to keep this PR diff smaller while we determine the best naming convention to follow throughout modules (`Cardinal` public naming convention will not stay - just kept the name of the existing class made public for now)

 ### Checklist

 - [x] Added a changelog entry

### Authors

- @sarahkoop

### CODE REVIEW

File: Demo/src/main/java/com/braintreepayments/demo/CardFragment.java [hunk 1]
```diff
@@ -178,7 +189,8 @@ public void onPurchase(View v) {
         card.setExpirationMonth(cardForm.getExpirationMonth());
         card.setExpirationYear(cardForm.getExpirationYear());
         card.setCvv(cardForm.getCvv());
-        card.setShouldValidate(false); // TODO GQL currently only returns the bin if validate = false
+        card.setShouldValidate(
+                false); // TODO GQL currently only returns the bin if validate = false
```
Comment (sshropshire) [inline Demo/src/main/java/com/braintreepayments/demo/CardFragment.java:193]:
```suggestion
        // TODO: GQL currently only returns the bin if validate = false
        card.setShouldValidate(false); 
```

File: ThreeDSecure/src/main/java/com/braintreepayments/api/CardinalResult.java [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 import com.cardinalcommerce.cardinalmobilesdk.models.ValidateResponse;
 
-class CardinalResult {
+public class CardinalResult {
```
Comment (sshropshire) [inline ThreeDSecure/src/main/java/com/braintreepayments/api/CardinalResult.java:5]: q: Can this remain `package-private` or is it required after refactoring.

File: ThreeDSecure/src/main/java/com/braintreepayments/api/CardinalResultCallback.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.braintreepayments.api;
+
+import androidx.annotation.NonNull;
+
+public interface CardinalResultCallback {
+
+    void onCardinalResult(@NonNull CardinalResult cardinalResult);
```
Comment (sshropshire) [inline ThreeDSecure/src/main/java/com/braintreepayments/api/CardinalResultCallback.java:7]: Oh wait nevermind, I see now.

File: ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureClient.java [hunk 4]
```diff
@@ -35,90 +27,28 @@ public class ThreeDSecureClient {
...
-     * <p>
-     * Deprecated. Use {@link ThreeDSecureClient(Fragment, BraintreeClient)} or
-     * {@link ThreeDSecureClient(FragmentActivity, BraintreeClient)} instead.
      *
      * @param braintreeClient a {@link BraintreeClient}
      */
-    @Deprecated
     public ThreeDSecureClient(@NonNull BraintreeClient braintreeClient) {
-        this(null, null, braintreeClient, new CardinalClient(),
+        this(braintreeClient, new CardinalClient(),
```
Comment (sshropshire) [inline ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureClient.java:37]: 🥹

**Review by sshropshire — APPROVED** [review]
🙌

**Comment by sshropshire** [comment]
This is somewhat off topic, but I always found it odd that [ThreeDSecureInfo.java](https://github.com/braintree/braintree_android/blob/7949b4c6a3ffca1ff05579154ef5401fee3dde2f/Card/src/main/java/com/braintreepayments/api/ThreeDSecureInfo.java) is located in the `:Card` module. Wondering if maybe we have a chance to fix this in `v5`?

**Comment by sarahkoop** [comment]
> This is somewhat off topic, but I always found it odd that [ThreeDSecureInfo.java](https://github.com/braintree/braintree_android/blob/7949b4c6a3ffca1ff05579154ef5401fee3dde2f/Card/src/main/java/com/braintreepayments/api/ThreeDSecureInfo.java) is located in the `:Card` module. Wondering if maybe we have a chance to fix this in `v5`?

We have a ticket to create a `ThreeDSecureNonce` class in v5 which should resolve that issue - it'll mean we aren't exactly aligned with iOS but I think it's a good move 👍

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**